From: Dennis Dalessandro <dennis.dalessandro@intel.com>
Subject: IB/rdmavt: Remove RVT_FLAGs
Git-commit: ce73fe25a6bd4a4deabed57e2553100e10fb8362
Patch-mainline: v4.6
References: bsc#973818, fate#319242
Acked-by: Jiri Bohac <jbohac@suse.cz>

While hfi1 and qib were still supporting bits and pieces of core verbs
components there needed to be a way to convey if rdmavt should handle
allocation and initialize of resources like the queue pair table. Now
that all of this is moved into rdmavt there is no need for these flags.
They are no longer used in the drivers.

Reviewed-by: Ira Weiny <ira.weiny@intel.com>
Reviewed-by: Jubin John <jubin.john@intel.com>
Signed-off-by: Dennis Dalessandro <dennis.dalessandro@intel.com>
Signed-off-by: Doug Ledford <dledford@redhat.com>
---
 drivers/infiniband/sw/rdmavt/cq.c |    5 -----
 drivers/infiniband/sw/rdmavt/mr.c |    5 -----
 drivers/infiniband/sw/rdmavt/qp.c |    8 --------
 include/rdma/rdma_vt.h            |   10 ----------
 4 files changed, 28 deletions(-)

--- a/drivers/infiniband/sw/rdmavt/cq.c
+++ b/drivers/infiniband/sw/rdmavt/cq.c
@@ -508,11 +508,6 @@ int rvt_driver_cq_init(struct rvt_dev_in
 	int cpu;
 	struct task_struct *task;
 
-	if (rdi->flags & RVT_FLAG_CQ_INIT_DRIVER) {
-		rvt_pr_info(rdi, "Driver is doing CQ init.\n");
-		return 0;
-	}
-
 	if (rdi->worker)
 		return 0;
 	rdi->worker = kzalloc(sizeof(*rdi->worker), GFP_KERNEL);
--- a/drivers/infiniband/sw/rdmavt/mr.c
+++ b/drivers/infiniband/sw/rdmavt/mr.c
@@ -66,11 +66,6 @@ int rvt_driver_mr_init(struct rvt_dev_in
 	unsigned lk_tab_size;
 	int i;
 
-	if (rdi->flags & RVT_FLAG_MR_INIT_DRIVER) {
-		rvt_pr_info(rdi, "Driver is doing MR init.\n");
-		return 0;
-	}
-
 	/*
 	 * The top hfi1_lkey_table_size bits are used to index the
 	 * table.  The lower 8 bits can be owned by the user (copied from
--- a/drivers/infiniband/sw/rdmavt/qp.c
+++ b/drivers/infiniband/sw/rdmavt/qp.c
@@ -173,11 +173,6 @@ int rvt_driver_qp_init(struct rvt_dev_in
 	int i;
 	int ret = -ENOMEM;
 
-	if (rdi->flags & RVT_FLAG_QP_INIT_DRIVER) {
-		rvt_pr_info(rdi, "Driver is doing QP init.\n");
-		return 0;
-	}
-
 	if (!rdi->dparms.qp_table_size)
 		return -EINVAL;
 
@@ -284,9 +279,6 @@ void rvt_qp_exit(struct rvt_dev_info *rd
 	if (!rdi->qp_dev)
 		return;
 
-	if (rdi->flags & RVT_FLAG_QP_INIT_DRIVER)
-		return; /* driver did the qp init so nothing else to do */
-
 	kfree(rdi->qp_dev->qp_table);
 	free_qpn_table(&rdi->qp_dev->qpn_table);
 	kfree(rdi->qp_dev);
--- a/include/rdma/rdma_vt.h
+++ b/include/rdma/rdma_vt.h
@@ -60,16 +60,6 @@
 #include <rdma/rdmavt_mr.h>
 #include <rdma/rdmavt_qp.h>
 
-/*
- * For some of the IBTA objects there will likely be some
- * initializations required. We need flags to determine whether it is OK
- * for rdmavt to do this or not. This does not imply any functions of a
- * partiuclar IBTA object are overridden.
- */
-#define RVT_FLAG_MR_INIT_DRIVER BIT(1)
-#define RVT_FLAG_QP_INIT_DRIVER BIT(2)
-#define RVT_FLAG_CQ_INIT_DRIVER BIT(3)
-
 #define RVT_MAX_PKEY_VALUES 16
 
 struct rvt_ibport {
